src/test/java/net/snowflake/ingest/FipsCompliantIngestIT.java

```diff
@@ -0,0 +1,328 @@
+        getFileName().toString());
+    files.add(IngestExampleHelper.createTempCsv(filesDirectory, "sample", 15).
+        getFileName().toString());
+    files.add(IngestExampleHelper.createTempCsv(filesDirectory, "sample", 20).
+        getFileName().toString());
+    files.add(fileWithWrongCSVFormat);
+    conn = IngestExampleHelper.getConnection(USER, PASSWORD,
+                                             ACCOUNT, HOST, PORT);
+    keypair = generateKeyPairWithBCFipsProvider();
+    manager = new SimpleIngestManager(ACCOUNT, USER,
+                                      FQ_PIPE, keypair, SCHEME,
+                                      HOST, PORT);
+    setup(files, filesLocationUrl);
+  }
+
+  /**
+   * Remove test table and pipe
+   */
+  @After
+  public void afterAll()
+  {
+    IngestExampleHelper.doQuery(conn, "drop pipe if exists " + FQ_PIPE);
+    IngestExampleHelper.doQuery(conn, "drop stage if exists " + STAGE);
+    IngestExampleHelper.doQuery(conn, "drop table if exists " + TABLE);
+  }
+
+  @Test
+  public void testIngestWithFipsComplianceMode() throws Exception
+  {
+    System.out.println("Starting snowflake ingest client");
+    System.out.println("Connecting to " + SCHEME + "://" +
+                           HOST + ":" + PORT +
+                           "\n with Account:" + ACCOUNT +
+                           ", User: " + USER);
+
+    try
+    {
+      final long oneHourMillis = 1000 * 3600l;
+      String startTime = Instant
+          .ofEpochMilli(System.currentTimeMillis() - 4 * oneHourMillis).toString();
+
+      IngestResponse insertFileResponse = insertFiles(files);
+      System.out.println("Received ingest insertFileResponse: " + insertFileResponse.toString());
+
+      Assert.assertEquals("SUCCESS", insertFileResponse.getResponseCode());
+
+      HistoryResponse history = waitForFilesHistory(files);
+      System.out.println("Received history response: " + history.toString());
+      String endTime = Instant
+          .ofEpochMilli(System.currentTimeMillis()).toString();
+
+      HistoryRangeResponse historyRangeResponse =
+          manager.getHistoryRange(null,
+                                  startTime,
+                                  endTime);
+
+      System.out.println("Received history range response: " +
+                             historyRangeResponse.toString());
+      // Lets iterate over the historyResponse and add all rowsInserted
+
+      final long actualSumOfRowsInserted = historyRangeResponse.files
+          .stream()
+          .map((HistoryResponse.FileEntry::getRowsInserted))
+          .mapToLong(Long::longValue)
+          .sum();
+
+      final Set<String> loadedStatus = historyRangeResponse.files
+          .stream()
+          .map(HistoryResponse.FileEntry::getStatus)
+          .map(IngestStatus::toString)
+          .collect(Collectors.toSet());
+
+      // Sum of all rows = 45
+      Assert.assertEquals(45, actualSumOfRowsInserted);
```

Comment:
Why 45? Would be better to use constants and derive this number, rather than have a "magic value" for it :)